draft-ietf-mls-protocol.md

```diff
@@ -1353,32 +1365,32 @@ Given these inputs, the derivation of secrets for an epoch
 proceeds as shown in the following diagram:
 
 ~~~~~
-               init_secret_[n-1] (or 0)
-                     |
-                     V
-    PSK (or 0) -> HKDF-Extract = early_secret
-                     |
-               Derive-Secret(., "derived", "")
-                     |
-                     V
-commit_secret -> HKDF-Extract = epoch_secret
+ init_secret_[n-1] (or 0)-+
+                          |
+   commit_secret-----+    |
+                     |    |
+     PSK (or 0)-+    |    |
+                |    |    |
+                |    |    |
+                V    V    V
+               Derive-Secrets([., ., .], "combine", unique_value) = epoch_secret
                      |
                      +--> HKDF-Expand(., "mls 1.0 welcome", Hash.length)
                      |    = welcome_secret
                      |
-                     +--> Derive-Secret(., "sender data", GroupContext_[n])
+                     +--> Derive-Secret(., "sender data", (unique_value, GroupContext_[n]))
```

Comment:
The `(x, y)` syntax is not defined.  And in any case, I think `GroupContext_[n]` is the unique value you're looking for.

Reply:
(x,y) should refer to an injective encoding of pairs. If GroupContext_[n] is already unique, this simplifies the construction and would be very nice. We were hoping that GroupContext_[n] is already unique, but we don't have the full MLS context to make this judgement. Is GroupContext_[n] also already available at the point when we compute the epoch_secret?

Reply:
Yes GroupContext_[n] is what you're looking for, and I also don't like the tuples.

Reply:
Thanks for the clarification.  We would want a defined injective encoding here.  But if we agree that GroupContext is unique enough, maybe we can just use that?